### PR TITLE
add more test cases

### PR DESCRIPTION
**Pull request by inovizz**
- migrate all available test cases from LMS to upgradable contract
- remove the testLMS.js file, since test cases from that file would not be required now.
- add a generic Event in DataStore.sol which can be used by Borrow, Return and Rate functionality
- optimize logic for OnlyMember modifier in Organisation.sol

### CODE REVIEW

File: contracts/DataStore.sol [hunk 1]
```diff
@@ -11,6 +11,12 @@ contract DataStore is Killable {
     // The data in this contract can be changed only by the owner, which should be the calling contract.
     uint public count;
 
+    event Event(uint indexed id, address indexed account, uint timestamp, string key, string comments, uint indexed rating);
```
Comment (nrchandan) [inline contracts/DataStore.sol]: The argument names will stand out as foreigners in this contract. May be use generic names?

File: contracts/Organisation.sol [hunk 2]
```diff
@@ -144,14 +141,20 @@ contract Organisation is Ownable {
 
     function borrowBook(uint id) payable onlyMember {
         bookStore.borrowBook(id);
+        var key = "borrow";
+        DataStore(bookStore).triggerEvent(id, msg.sender, "", 0, key);
```
Comment (nrchandan) [inline contracts/Organisation.sol]: Can we define a wrapper triggerEvent function in BooksLibrary and calling bookStore.triggerEvent?

Comment (nrchandan) [inline contracts/Organisation.sol]: What's the thought behind defining they "key" variable?

File: contracts/DataStore.sol (hunk 1 above)
Comment (inovizz) [inline contracts/DataStore.sol]: Done.

File: contracts/Organisation.sol (hunk 2 above)
Comment (inovizz) [inline contracts/Organisation.sol]: Made the suggested changes.

File: contracts/BooksLibrary.sol [hunk 3]
```diff
@@ -110,6 +106,20 @@ library BooksLibrary {
 
         // All reviews are logged. Applications are responsible for eliminating duplicate ratings
         // and computing average rating.
-//        Rate(id, msg.sender, rating, comments, now);
+    }
+
+    function borrowEvent(address bookStoreAddress, uint id, address account) {
+        var bookStore = DataStore(bookStoreAddress);
+        bookStore.triggerEvent(id, account, "", 0, "borrow");
```
Comment (nrchandan) [inline contracts/BooksLibrary.sol]: Would it make more sense to have the event type ("borrow", "return"...) as the first argument?

File: contracts/DataStore.sol [hunk 4]
```diff
@@ -11,6 +11,12 @@ contract DataStore is Killable {
     // The data in this contract can be changed only by the owner, which should be the calling contract.
     uint public count;
 
+    event Event(uint indexed index, address indexed account,string value, uint indexed number, string key, uint timestamp);
```
Comment (nrchandan) [inline contracts/DataStore.sol]: Would "eventName" be more expressive than "key"? Should we have "eventName" as the first argument?

Comment (nrchandan) [inline contracts/DataStore.sol]: Missing a space after comma.

File: contracts/Organisation.sol [hunk 5]
```diff
@@ -144,14 +141,17 @@ contract Organisation is Ownable {
 
     function borrowBook(uint id) payable onlyMember {
         bookStore.borrowBook(id);
+        bookStore.borrowEvent(id, msg.sender);
```
Comment (nrchandan) [inline contracts/Organisation.sol]: Would it make sense to invoke the borrowEvent function from the library itself? Same for others.

Comment (inovizz) [inline contracts/Organisation.sol]: How about I call the triggerEvent function from borrowBook, returnBook, and rateBook itself?
With this, we don't need to define separate functions for triggering events.

File: contracts/BooksLibrary.sol (hunk 3 above)
Comment (inovizz) [inline contracts/BooksLibrary.sol]: ok sure.

File: contracts/DataStore.sol (hunk 4 above)
Comment (inovizz) [inline contracts/DataStore.sol]: Agree.

File: contracts/Organisation.sol (hunk 5 above)
Comment (nrchandan) [inline contracts/Organisation.sol]: Yeah, that would work too.

**Pull request merged**